addons/account/models/account_journal.py

```diff
@@ -600,8 +600,18 @@ def _alias_prepare_alias_name(self, alias_name, name, code, jtype, company):
         )
         if company != self.env.ref('base.main_company'):
             company_identifier = company.name if self.env['mail.alias']._is_encodable(company.name) else company.id
-            if f'-{company_identifier}' not in alias_name:
-                alias_name = f"{alias_name}-{company_identifier}"
+            if f'-{company_identifier}' not in base_alias_name:
+                base_alias_name = f'{base_alias_name}-{company_identifier}'
+
+        # Check for duplicate alias, if exists append a suffix to ensure uniqueness
+        # It's not a duplicate if it's the same journal
+        sanitized_alias_name = self.env['mail.alias']._sanitize_alias_name(base_alias_name)
+        existing_alias = self.env['mail.alias'].search([('alias_name', '=', sanitized_alias_name)])
+        if len(existing_alias) > 0 and existing_alias.id != self.alias_id.id:
+            alias_name = f'{sanitized_alias_name}-{code}'
+        else:
+            alias_name = sanitized_alias_name
```

Comment:
```suggestion
        existing_alias = self.env['mail.alias'].search([('alias_name', '=', sanitized_alias_name), ('id', '!=', self.alias_id.id)])
        alias_name = f'{sanitized_alias_name}-{code}' if existing_alias else sanitized_alias_name
        return self.env['mail.alias']._sanitize_alias_name(alias_name)
``` 
Something like this is a bit cleaner IMO, but you seem to have forgotten about TDE's comment on the uniqueness of alias name inside an alias domain. 

Also, I see the function you try to improve is defined with a `api.model`, meaning that the content of the self should not be relevant. However, you now use self.alias_id.id.